public/modules/custom/grants_handler/src/Plugin/WebformHandler/GrantsHandler.php

```diff
@@ -376,8 +375,31 @@ public function prepareForm(WebformSubmissionInterface $webform_submission, $ope
     // probably will change when we have proper company selection process.
     $selectedCompany = $this->grantsProfileService->getSelectedCompany();
 
-    $grantsProfileDocument = $this->grantsProfileService->getGrantsProfile($selectedCompany['identifier']);
-    $grantsProfile = $grantsProfileDocument->getContent();
+    if ($selectedCompany == NULL) {
+      throw new CompanySelectException('User not authorised');
+    }
+
+    try {
+      $grantsProfileDocument = $this->grantsProfileService->getGrantsProfile($selectedCompany['identifier']);
+      if (gettype($grantsProfileDocument) == 'object' && get_class($grantsProfileDocument) == 'Drupal\helfi_atv\AtvDocument') {
+        $grantsProfile = $grantsProfileDocument->getContent();
+      }
+      else {
+        $this->messenger()->addWarning('You must have grants profile created.');
+
+        $url = Url::fromRoute('grants_profile.edit');
+        $redirect = new RedirectResponse($url->toString());
+        $redirect->send();
```

Comment:
Some duplicated code here, maybe we can just throw the exception since it has the same code

```suggestion
        throw new \Exception();
```